### ISSUE TITLE
Databricks View doesn't support long_lines ignore_comment_clauses setting

### ISSUE THREAD

**Issue by avinmathew** [sqlfluff/sqlfluff#7992]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The ignore_comment_clauses = True for the layout.long_lines rule (LT05) doesn't work with a Databricks syntax CREATE VIEW with a COMMENT. When running a lint, it fails with a "Line is too long" error.

### Expected Behaviour

The `ignore_comment_clauses = True` should skip the rule for COMMENT clauses within CREATE VIEW statements.

### Observed Behaviour

I received the error:
```
L:   2 | P:   1 | LT05 | Line is too long (654 > 100).
                       | [layout.long_lines]
```

### How to reproduce

Create a test_view.sql with:

```sql
CREATE VIEW my_schema.my_view
COMMENT 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer nec dignissim diam, eu consectetur dolor. Aliquam vestibulum hendrerit massa ac dapibus. Duis luctus ut nunc quis mollis. Pellentesque at ultricies justo. Mauris nulla metus, posuere in lorem eget, mattis eleifend dui. In ullamcorper, enim id posuere mattis, urna ex tempus ipsum, ac bibendum neque arcu at mi. Pellentesque ultricies rutrum nibh a accumsan. Morbi non fermentum eros. Sed at nisl et purus suscipit congue. Vestibulum auctor vehicula maximus. In semper felis neque, eu condimentum lacus pharetra ut. Curabitur nibh metus, dapibus sed tellus ac, commodo porta orci.'
AS
SELECT 1 AS col
```

Set .sqlfluff to:
```
[sqlfluff]
dialect = databricks
max_line_length = 100

[sqlfluff:rules:layout.long_lines]
ignore_comment_clauses = True
```

Run `sqlfluff lint test_view.sql`.

The error should be raised.

Functions operate correctly and no errors are raised.
```sql
CREATE FUNCTION my_schema.my_function ()
RETURNS TABLE
COMMENT 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer nec dignissim diam, eu consectetur dolor. Aliquam vestibulum hendrerit massa ac dapibus. Duis luctus ut nunc quis mollis. Pellentesque at ultricies justo. Mauris nulla metus, posuere in lorem eget, mattis eleifend dui. In ullamcorper, enim id posuere mattis, urna ex tempus ipsum, ac bibendum neque arcu at mi. Pellentesque ultricies rutrum nibh a accumsan. Morbi non fermentum eros. Sed at nisl et purus suscipit congue. Vestibulum auctor vehicula maximus. In semper felis neque, eu condimentum lacus pharetra ut. Curabitur nibh metus, dapibus sed tellus ac, commodo porta orci.'
RETURN
SELECT 1 AS col
```

### Dialect

Databricks

### Version

sqlfluff, version 4.2.1
Python 3.11.9

### Configuration

[sqlfluff]
dialect = databricks
max_line_length = 100

[sqlfluff:rules:layout.long_lines]
ignore_comment_clauses = True

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
